### ISSUE TITLE
Distorted video after skipping to next or prev chapter

### ISSUE THREAD

**Issue by amverweij** [xbmc/xbmc#19294]
### Describe the bug
Here is a clear and concise description of what the problem is:

When I play a movie on Kodi, it plays perfectly fine. Pause/resume also works as expected. However, when I skip to the next or previous chapter, the image becomes distorted.
 
I have this issue with movies that are encoded in x264 and x265. I consider it a bug because it makes it impossible to start watching a movie from any other point then the beginning, or to watch a scene again.

Similar issues occur when seeking using fast forward or rewind.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

I expect the video to be rendered correctly, also after I skip to the next or previous chapter.

## Actual Behavior

After skipping to the next or the previous chapter, the display becomes garbled.

### To Reproduce

1.  Start kodi-gbm (from https://github.com/popcornmix/xbmc/tree/gbm) on a 64-bit RPi-4B
2.  Play any x264 encoded movie that contains chapters
3.  Press "next" on the remote to skip to the next chapter

### Debuglog

The debuglog can be found here:

https://pastebin.com/WxtuUk6i

### Screenshots 
Here are some links or screenshots to help explain the problem:

https://chet.dnsalias.com/screenshots/20210226_162019.jpg
https://chet.dnsalias.com/screenshots/20210226_162025.jpg
https://chet.dnsalias.com/screenshots/20210226_162027.jpg

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

I am trying to use kodi-gbm on a RPi-4B under Gentoo Linux (64-bit).

I have tried to reproduce these issues using a LibreElec nightly build (LibreELEC-RPi4.arm-9.80-nightly-20210226-e12ff43.img). Using LibreElec, skipping to the next and/or previous chapter, as well as fast forward/rewind all work as expected. So I'm assuming it is a 64-bit issue.

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Linux herbie 5.10.14-v8+ #2 SMP PREEMPT Fri Feb 12 19:30:48 CET 2021 aarch64 GNU/Linux
 - Kodi version: 19.0, kodi-gbm + the 12 commits from popcornmix's gbm branch.

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by amverweij** [comment]
After rebuilding the system in 32 bit, I am still able to reproduce.

Linux herbie 5.10.17-v7l+ #1403 SMP Mon Feb 22 11:33:35 GMT 2021 armv7l ARMv7 Processor rev 3 (v7l) BCM2711 GNU/Linux

**Comment by amverweij** [comment]
Long story short, I can't reproduce anymore.

My RPi4-B PXE boots into a nbd-root. I made the "screenshots" on my 64-bit ndb root device.

Since the LibreElec nightly build works fine, I tried to eliminate all possible differences, the first being switching to 32-bit. For this I prepared a separate nbd block device, leaving the 64-bit one as it was.

I could reproduce the issue with my first 32-bit builds. After that I started eliminating differences with LibreElec one by one, and testing the effect. Not using (suitable patched) system ffmpeg, compiler flags, etc. At some point my 32-bit nbd-root started working, like LIbreElec (this was when switching the compiler target to cortex-a53 instead if cortex-a72). However when rolling back my changes in order to get a working reproduction scenario, things kept working. And to my big surprise, I cannot reproduce anymore in the 64 bit nbd root environment either.

The two environments share /boot and /home, so I guess something must have changed in either of those, however I can't reproduce what did.

Anyways, things are working as expected. Thanks to all the devs for making this great software! Much appreciated.
